cmd/vicadmin/vicadm.go

```diff
@@ -140,6 +160,37 @@ func configureReaders() []entryReader {
 	return readers
 }
 
+type authenticator interface {
```

Comment:
@fdawg4l you know I was asking myself the same question and erred on the side of private as I wasn't sure it *could* be used elsewhere.. but now that I think about it, anywhere that we're registering HTTP route handlers with stdlib `http.HandlerFunc` it should be useful.. I could move it to a `pkg/`..